CMakeLists.txt

```diff
@@ -75,5 +75,6 @@ endif()
 
 add_subdirectory(src)
 add_subdirectory(tests)
+add_subdirectory(config/tools)
```

Comment:
I don't know how I missed this, but why is the configuration-generator in its own directory rather than in `tools/config`? It makes perfect sense (to me) to have the generated configuration files (as well as the ones we ship pre-made) in a unified directory, but we have a `tools` directory specifically to handle ancillary tooling (just like this).

Reply:
Well, i created the Config directory to organize the configuration files around the repo. It just wouldn't have made sense, i thought, to have the config files in tools/Config. But to have them in Config and to have the config generator in config/tools seemed to make more sense. Would you rather change it? And if so, do you want the config files in tools/config as well? That would seem a bit odd to me

Reply:
Sorry, to clarify, I meant that having `/config` for the configuration files makes perfect sense. But having the configuration-generator live outside `/tools` (which is the specific place for ancillary tooling) doesn't. I'd expect `/tools/config` to have the generator and `/config` to house the pre-made configurations.

Does that feel like a reasonable breakdown?

Reply:
That was my original plan. But i remember, i think i had asked you on a call, or somewhere, if it should go in tools and you seemed apprehensive. I will put it there though

Reply:
Mmm, my apologies for steering you wrong. But, having the tool sit somewhere under `/tools` and the pre-made configs under `/config` sounds ideal to me.

Reply:
DONE